examples/batch/batch.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	pgx "github.com/jackc/pgx/v5"
+	pgconn "github.com/jackc/pgx/v5/pgconn"
+	pgxpool "github.com/jackc/pgx/v5/pgxpool"
+)
+
+type PgxIface interface {
+	Begin(context.Context) (pgx.Tx, error)
+	Exec(context.Context, string, ...interface{}) (pgconn.CommandTag, error)
+	Close()
+}
+
+func databaseSetup(db PgxIface) (err error) {
+	// Create a new table 'ledger'
+	sql := `CREATE TABLE IF NOT EXISTS ledger (
+		id BIGINT PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+		description TEXT NOT NULL,
+		amount BIGINT NOT NULL);`
+	_, err = db.Exec(context.Background(), sql)
+	return err
+}
+
+func iterateResults(br pgx.BatchResults, QueuedQueries []*pgx.QueuedQuery) error {
+
+	file, err := os.Create("OUTPUT.md")
+	if err != nil {
+		return fmt.Errorf("iterateResults: %s", err)
+	}
+	defer file.Close()
+
+	fmt.Fprintf(file, "## PostgreSQL Batch Example output\n")
+
+	// Iterate over a batch of queued queries
+	for _, query := range QueuedQueries {
+
+		// Print SQL field of the current query
+		fmt.Fprintf(file, "### %v \n", query.SQL)
+
+		//  reads results from the current query
+		rows, err := br.Query()
+		if err != nil {
+			return fmt.Errorf("iterateResults: %s", err)
+		}
+
+		// Print column headers
+		fmt.Fprintf(file, "- *DESCRIPTION* , *AMOUNT* \n")
+		// Iterate over the resulted rows
+		//
+		var id, amount, descr = int64(0), int64(0), string("")
+		_, err = pgx.ForEachRow(rows, []any{&id, &descr, &amount}, func() error {
+			fmt.Fprintf(file, "- \"%v\" , %d \n", descr, amount)
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("iterateResults: %s", err)
+		}
+	}
+
+	return err
+}
+
+func requestBatch(db PgxIface) (err error) {
+
+	// Initialize a database object
+	tx, err := db.Begin(context.Background())
+	if err != nil {
+		return fmt.Errorf("requestBatch: %s", err)
+	}
```

Comment:
Why do you think we need to start a new transaction here? Batch is executed within a single transaction anyway. Do you plan to add other statements?